server/src/main/java/org/elasticsearch/action/synonyms/PutSynonymsAction.java

```diff
@@ -61,7 +62,11 @@ public Request(StreamInput in) throws IOException {
 
         public Request(String synonymsSetId, BytesReference content, XContentType contentType) throws IOException {
             this.synonymsSetId = synonymsSetId;
-            this.synonymRules = PARSER.apply(XContentHelper.createParser(XContentParserConfiguration.EMPTY, content, contentType), null);
+            try (XContentParser parser = XContentHelper.createParser(XContentParserConfiguration.EMPTY, content, contentType)) {
+                this.synonymRules = PARSER.apply(parser, null);
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Failed to parse: " + content.utf8ToString(), e);
```

Comment:
Should the content we're returning in the log message be truncated here if it's too long?